98_exercises.Rmd

```diff
@@ -0,0 +1,78 @@
+
+1. Import data
+2. Construct a TreeSE.
+3. Check that importing is done correctly. E.g., you can choose random samples and features,
+and check that their values equal between raw files and TreeSE.
+
+Functions that might be useful: DataFrame, TreeSummarizedExperiment, matrix, rownames, colnames, SimpleList
+
+## Data exploration
+
+1. Please install the latest development version of mia from GitHub.
+2. Load experimental dataset from mia.
+3. What are the dimensions? (How many samples there are, and how many taxa in each taxonomic rank)?
+4. Calculate library size.
+5. Summarize sample metadata variables. (How many age groups, how they are distributed? 0%, 25%, 50%, 75%, and 100% quantiles of library size?)
+6. Create two histograms. Another shows the distribution of absolute counts, another shows how CLR transformed values are distributed.
+7. Visualize how relative abundances are distributed between taxa in samples.
+8. Calculate Shannon diversity index. Is it significantly different between vegan and mixed diet?
+
+Functions that might be useful: nrow, ncol, dim, summary, table, quantile, unique, transformSamples, ggplot, wilcox.test, addPerCellQC, agglomerateByRank, estimateDiversity, plotAbundance
+
+## Differential abundance analysis
+
+1. Please install the latest development version of mia from GitHub.
+2. Load experimental dataset from mia.
+3. Compare abundances of each taxa between groups. First, use Wilcoxon or Kruskall-Wallis test. Then use some other method dedicated to microbiome data. 
+4. Summarize findings by plotting a taxa vs samples heatmap. Add column annotation that tells the group of each sample, and row annotation that tells whether the difference of certain taxa was statistically significant.
+5. Choose statistically significant taxa and visualize their abundances with boxplot & jitterplot.
+
+Functions that might be useful: wilcox.test, kruskal.test, ggplot, pheatmap, ComplexHeatMap::Heatmap, ancombc, aldex2, maaslin2, agglomerateByRank, transformSamples, transformFeatures, subsetByPrevalentTaxa
+
+## Basics of community composition
+
+1. Please install the latest development version of mia from GitHub.
+2. Load experimental dataset from mia.
+3. Create a PCoA with Aitchison dissimilarities. How much coordinate 1 explains the differences? How about coordinate 2?
+4. Create dbRDA with Bray-Curtis dissimilarities on relative abundances. Use PERMANOVA. Can differences between samples be explained with variables of sample meta data? 
+5. Analyze diets' association on beta diversity. Calculate dbRDA and then PERMANOVA. Visualize coefficients. Which taxa's abundances differ the most between samples? 
+6. Interpret your results. Is there association between community composition and location? What are those taxa that differ the most; find information from literature.
+
+Functions that might be useful: runMDS, runRDA, anova.cca, transformSamples, agglomerateByRank, ggplot, plotReducedDim
```

Comment:
Useful functions could be potentially listed after each numbered task in parentheses (but good like this already)

Reply:
I'm wandering if that is too easy. Googling and searching from OMA could be part of the task (like in real life)

referring also hyperlinks